File: 1561.maximum-number-of-coins-you-can-get.py
#
# @lc app=leetcode id=1561 lang=python3
#
# [1561] Maximum Number of Coins You Can Get
#

# @lc code=start
from typing import List


class Solution:
    # Sort piles
    # Then add up second last from the back up to n // 3
    def maxCoins(self, piles: List[int]) -> int:
        n = len(piles)
        piles.sort()
        result = 0
        for i in range(n-2, n // 3 -1, -2):
            result += piles[i]
        return result
        
# @lc code=end

File: 1561.maximum-number-of-coins-you-can-get.cs
/*
 * @lc app=leetcode id=1561 lang=csharp
 *
 * [1561] Maximum Number of Coins You Can Get
 */

// @lc code=start
public class Solution 
{
    public int MaxCoins(int[] piles)
    {
        int n = piles.Length;
        int result = 0;
        Array.Sort(piles);
        for (int i = n - 2; i > n / 3 - 1; i-=2)
        {
            result += piles[i];
        }
        return result;
    }
}
// @lc code=end

